problem-spring-webflux/src/main/java/org/zalando/problem/spring/webflux/advice/security/AuthenticationAdviceTrait.java

```diff
@@ -26,6 +26,9 @@
     @ExceptionHandler
     default Mono<ResponseEntity<Problem>> handleAuthentication(final AuthenticationException e,
             final ServerWebExchange request) {
+        if (e instanceof AuthenticationServiceException) {
```

Comment:
Shouldn't Spring do this for us automatically because `AuthenticationServiceException` is more specific?

Reply:
Ideally it would be done by the resolver (or whatever the correct term is).

Reply:
I couldn't find a way to do it. See RequestMappingHandlerAdapter::handleException for how exceptions are handled in WebFlux.